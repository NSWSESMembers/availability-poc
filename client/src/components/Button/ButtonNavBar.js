import React from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

import styles from './styles';

const ButtonNavBar = ({ icon, onPress }) => (
  <TouchableOpacity onPress={() => onPress()}>
    <Icon name={icon} size={24} style={styles.buttonNavBarIcon} />
  </TouchableOpacity>
);

ButtonNavBar.propTypes = {
  icon: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

export default ButtonNavBar;
