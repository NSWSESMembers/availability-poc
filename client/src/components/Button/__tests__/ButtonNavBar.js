import React from 'react';
import renderer from 'react-test-renderer';

import ButtonNavBar from '../ButtonNavBar';

const noop = () => {};

test('renders correctly', () => {
  const tree1 = renderer.create(
    <ButtonNavBar
      icon="glass"
      onPress={noop}
    />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();
});
