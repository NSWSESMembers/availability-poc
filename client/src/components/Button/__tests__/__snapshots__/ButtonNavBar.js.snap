// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 24,
        },
        Object {
          "color": "#ffffff",
          "marginRight": 10,
        },
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ï€€
  </Text>
</View>
`;
